netspeed/src/netspeed.c

```diff
@@ -1145,104 +1160,126 @@ static void
-
-	/* signals */
-	gtk_builder_add_callback_symbols (builder,
-	                                  "on_network_device_combo_changed", G_CALLBACK (device_change_cb),
-	                                  "on_show_sum_checkbutton_toggled", G_CALLBACK (showsum_change_cb),
-	                                  "on_show_bits_checkbutton_toggled", G_CALLBACK(showbits_change_cb),
-	                                  "on_change_icon_checkbutton_toggled", G_CALLBACK (changeicon_change_cb),
-	                                  "on_show_icon_checkbutton_toggled", G_CALLBACK (showicon_change_cb),
-	                                  "on_show_quality_icon_checkbutton_toggled", G_CALLBACK (showqualityicon_change_cb),
-	                                  "on_preferences_dialog_response", G_CALLBACK(pref_response_cb),
-	                                  NULL);
-	gtk_builder_connect_signals (builder, applet);
-
-	g_object_unref (builder);
-
-	gtk_widget_show_all(GTK_WIDGET(applet->settings));
+    GtkBuilder *builder;
+    MateNetspeedApplet *applet = (MateNetspeedApplet*)data;
+    GList *ptr, *devices;
+    int i, active = -1;
+
+    g_assert (applet);
+
+    if (applet->settings)
+    {
+        gtk_window_present (GTK_WINDOW (applet->settings));
+        return;
+    }
+
+    builder = gtk_builder_new_from_resource (NETSPEED_RESOURCE_PATH "netspeed-preferences.ui");
+
+    applet->settings = GET_DIALOG ("preferences_dialog");
+    applet->network_device_combo = GET_WIDGET ("network_device_combo");
+
+    gtk_window_set_screen (GTK_WINDOW (applet->settings),
+                           gtk_widget_get_screen (GTK_WIDGET (applet->settings)));
+
+    gtk_dialog_set_default_response (GTK_DIALOG (applet->settings), GTK_RESPONSE_CLOSE);
+
+    g_settings_bind (applet->gsettings, "show-sum",
+                     gtk_builder_get_object (builder, "show_sum_checkbutton"),
+                     "active", G_SETTINGS_BIND_DEFAULT);
+
+    g_settings_bind (applet->gsettings, "show-bits",
+                     gtk_builder_get_object (builder, "show_bits_checkbutton"),
+                     "active", G_SETTINGS_BIND_DEFAULT);
+
+    g_settings_bind (applet->gsettings, "show-icon",
+                     gtk_builder_get_object (builder, "show_icon_checkbutton"),
+                     "active", G_SETTINGS_BIND_DEFAULT);
+
+    g_settings_bind (applet->gsettings, "show-quality-icon",
+                     gtk_builder_get_object (builder, "show_quality_icon_checkbutton"),
+                     "active", G_SETTINGS_BIND_DEFAULT);
+
+    g_settings_bind (applet->gsettings, "change-icon",
+                     gtk_builder_get_object (builder, "change_icon_checkbutton"),
+                     "active", G_SETTINGS_BIND_DEFAULT);
+
+    /* Default means device with default route set */
+    gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT (applet->network_device_combo),
+                                                        _("Default"));
```

Comment:
This indent is wrong, the `_("Default")` should be under the opening parenthesis of `gtk_combo_box_text_append_text`:
```
    gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT (applet->network_device_combo),
                                    _("Default"));
```